### PR TITLE
HOTT-3682-3402: Upload, download, create, expose CSV and XML files

### PR DESCRIPTION
**Pull request by maverickvi**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-3682
https://transformuk.atlassian.net/browse/HOTT-3402

### What?

I have added/removed/altered:

- [ ] Create and upload XML file
- [ ] Created files download API endpoint
- [ ] Expose files on backend with file_size and publication date
- [ ] Create ExchangeRateFile after uploading

### Why?

I am doing this because:

- We need to access CSV and XML files on the frontend

### Have you? (optional)

- [ ] Added documentation for new apis

### Deployment risks (optional)

- Includes destructive/migratory actions which may effect critical data

Screenshot 2023-07-27 at 15 34 36

### CODE REVIEW

**Comment by willfish** [comment]
This feels like a good time to move to using ActiveStorage

File: app/controllers/api/v2/exchange_rates/files_controller.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+module Api
+  module V2
+    module ExchangeRates
+      class FilesController < ApiController
+        def index
```
[the remark this answers is not in the document]
Comment (maverickvi) [reply app/controllers/api/v2/exchange_rates/files_controller.rb:5]: This endpoint is listing multiple items in those files and we need to pass more than 1 param so I thought index made more sense here

[the remark this answers is not in the document]
Comment (willfish) [reply app/controllers/api/v2/exchange_rates/files_controller.rb:5]: I guess it depends what you think the resource is :).

**Review by shjohnson — APPROVED** [review]

**Review by willfish — APPROVED** [review]

File: app/models/exchange_rates/period.rb [hunk 2]
```diff
@@ -17,6 +21,7 @@ def build(month, year)
         period = new
         period.month = month
         period.year = year
+        period.files = ExchangeRateFile.where(period_year: year, period_month: month)
```
[the remark this answers is not in the document]
Comment (shjohnson) [reply app/models/exchange_rates/period.rb:24]: FYI @maverickvi had a look at this and actually dont think is necessary or works quite how i was expecting.

[the remark this answers is not in the document]
Comment (willfish) [reply app/models/exchange_rates/period.rb:24]: It will add N+1s from my understanding

[the remark this answers is not in the document]
Comment (maverickvi) [reply app/models/exchange_rates/period.rb:24]: I think this will never be an issue. We have 12 months so 12 queries max. I'm marking this as resolved

**Pull request merged**